ios/PreferencesDB.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  PreferencesDB.swift
+//  magnet
+//
+//  Created by sam on 11/7/16.
+//  Copyright © 2016 Mozilla. All rights reserved.
+//
+
+import Foundation
+import SQLite
+
+class PreferencesDB {
+  private static var singleton: PreferencesDB!
+  private let table: Table = Table("preferences")
+  private let columnPrefKey = Expression<String>("pref_key");
+  private let columnPrefValue = Expression<String>("value");
+  private var db: Connection
+  
+  static func get() -> PreferencesDB {
+    let lockQueue = dispatch_queue_create("com.mozilla.magnet.preferences", nil);
+    
+    dispatch_sync(lockQueue) {
+      if (singleton == nil) {
+        singleton = PreferencesDB()
+      }
+    }
+    
+    return singleton
+  }
+  
+  private init() {
+    let path = NSSearchPathForDirectoriesInDomains(
+      .DocumentDirectory,
+      .UserDomainMask,
+      true).first!
+    
+    db = try! Connection("\(path)/history.sqlite3")
```

Comment:
For simple JSON data we should just use the `RequestStore`, although the term 'request' is a little misleading, it's really just our own key/value store, like `AsyncStorage`.